airbyte-integrations/connectors/source-mssql/src/testFixtures/java/io/airbyte/integrations/source/mssql/MsSqlTestDatabaseWithBackgroundThreads.java

```diff
@@ -281,6 +282,7 @@ public void close() {
       bgThread.stop = true;
     }
     super.close();
+    MssqlDebeziumStateUtil.disposeInitialState();
```

Comment:
Note: Because concurrent mssql tests are reusing threads in a fixed thread pool, it is necessary to dispose of cached state so the next test using this thread will start fresh.
Cached initial state is tied to a thread (`ThreadLocal<>`)